src/js/actions/libraries.js

```diff
@@ -0,0 +1,272 @@
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var Promise = require("bluebird"),
+        CCLibraries = require("cclibraries");
+
+    var descriptor = require("adapter/ps/descriptor"),
+        docAdapter = require("adapter/lib/document"),
+        libraryAdapter = require("adapter/lib/libraries");
+
+    var events = require("../events"),
+        locks = require("../locks");
+
+    /**
+     * Uploads the selected single layer to the current library
+     *
+     * Achieves this by:
+     *  - Creates a new element in the library
+     *  - Calls PS to export the layer to a temporary location
+     *  - Passes the temporary path to libraries API to update layer's content
+     *  - Tells Photoshop the location of the content
+     *  - Updates the document
+     *
+     * Eventually, we'll need this to accept layer(s), library, and be more flexible
+     * Also, we definitely need to get rid of the update document call, but this is 0.1
+     *
+     * @return {Promise}
+     */
+    var createElementFromSelectedLayer = function () {
+        var appStore = this.flux.store("application"),
+            libStore = this.flux.store("library"),
+            currentDocument = appStore.getCurrentDocument(),
+            currentLibrary = libStore.getCurrentLibrary(),
+            currentLayers = currentDocument.layers.selected;
+
+        if (!currentLibrary || currentLayers.count() !== 1) {
+            return Promise.resolve();
+        }
+
+        var currentLayer = currentLayers.first(),
+            IMAGE_ELEMENT_TYPE = "application/vnd.adobe.element.image+dcx",
+            REPRESENTATION_TYPE = "image/vnd.adobe.photoshop";
+
+        currentLibrary.beginOperation();
+
+        var newElement = currentLibrary.createElement(currentLayer.name, IMAGE_ELEMENT_TYPE),
+            representation = newElement.createRepresentation(REPRESENTATION_TYPE, "primary"),
+            previewSize = {
+                w: 248,
+                h: 188
+            };
+
+        // FIXME: Mac/Win temporary locations!
+        var exportObj = libraryAdapter.exportLayer("/tmp/", "/tmp/preview.png",
```

Comment:
Yikes. We'll need some way to get temporary paths... Might want to make a trello card for this too?